### ISSUE TITLE
Breaking Change with mailjet-apiv3-php

### ISSUE THREAD

**Issue by heresie** [mailjet/laravel-mailjet#16]
Hello there,

We are using the `mailjet/laravel-mailjet` dependency in our project, like this : 

```json
 {
     "require": {
         "mailjet/laravel-mailjet": "1.1"
     }
 }
```

Somewhere in our PHP files we are trying to send a new mail campaign, like this : 

```php
[...]

	public static function initCampaign($code_nl)
	{
		$call = [];
		$options = [];
		$mailjetService = new MailjetService(
			$GLOBALS['MJ_APIKEY_NL'][$code_nl]['PUBLIC'],
			$GLOBALS['MJ_APIKEY_NL'][$code_nl]['PRIVATE'],
			$call, $options
		);

		return new CampaignDraftService($mailjetService);
	}

[...]

	public function send() {
[...]
			$campaignDraftManager = self::initCampaign($newsletter->code_nl);
			$campaignDraft = new CampaignDraft('fr_FR', $newsletter->exp_nom, $newsletter->exp_email, $sujet, $idListeContacts);
			$campaignDraft->setOptionalProperties($optionalProp);
			$tabCompaign = $campaignDraftManager->create($campaignDraft);
			$content = [
				'Html-part' => $res,
				'Text-part' => 'Newsletter'
			];
				
			$campaignDraft->setContent($content);
			$campaignDraftManager->createDetailContent($tabCompaign[0]['ID'], $campaignDraft->getContent());
			$campaignDraftManager->sendCampaign($tabCompaign[0]['ID']);
[...]
```

All this code worked fine, since you pushed the 1.4 tag for the internal dependency `mailjet-apiv3-php`.

Since this repository has, in `composer.json` : 

```json
"mailjet/mailjet-apiv3-php": "^1.2",
```

As the new `1.4` tag is available, it will download `1.4` instead of `1.2` version.

In the `1.4` tag of `mailjet-apiv3-php`, the constructor for the `\Mailjet\Client` has changed, the method `_isBasicAuthenticationRequired` returns false, despite we sent to the service a key/secret combo variables (which we verified many times, are set and not empty). 

In this situation, it returns an error and fails to send any new campaign, because the `\Mailjet\Client` thinks we are sending only a token, and not the key/secret combo.

We fixed it by forcing the `1.2` tag of mailjet-apiv3-php in our `composer.json`.

Could you fix this by updating the `composer.json` file of the `1.1` tag and set the dependency for `mailjet-apiv3-php` to the `1.2` tag which works fine ?

**Comment by davidwter** [comment]
Same problem here, we had to force back to 1.2

**Comment by uavn** [comment]
fixed in 1.1.1 tag, mailjet-apiv3-php is locked to 1.2
